File: atlhyper_master_v2/model/service_account.go
// atlhyper_master_v2/model/service_account.go
// ServiceAccount Web API 响应类型（camelCase JSON tag，扁平结构）
package model

// ServiceAccountItem ServiceAccount 列表项
type ServiceAccountItem struct {
	Name                         string `json:"name"`
	Namespace                    string `json:"namespace"`
	SecretsCount                 int    `json:"secretsCount"`
	ImagePullSecretsCount        int    `json:"imagePullSecretsCount"`
	AutomountServiceAccountToken *bool  `json:"automountServiceAccountToken,omitempty"`
	CreatedAt                    string `json:"createdAt"`
	Age                          string `json:"age"`
}

// ServiceAccountDetail ServiceAccount 详情
type ServiceAccountDetail struct {
	Name                         string   `json:"name"`
	Namespace                    string   `json:"namespace"`
	SecretsCount                 int      `json:"secretsCount"`
	ImagePullSecretsCount        int      `json:"imagePullSecretsCount"`
	AutomountServiceAccountToken *bool    `json:"automountServiceAccountToken,omitempty"`
	SecretNames                  []string `json:"secretNames,omitempty"`
	ImagePullSecretNames         []string `json:"imagePullSecretNames,omitempty"`
	CreatedAt                    string   `json:"createdAt"`
	Age                          string   `json:"age"`

	Labels      map[string]string `json:"labels,omitempty"`
	Annotations map[string]string `json:"annotations,omitempty"`
}

File: atlhyper_web/src/api/service-account.ts
/**
 * ServiceAccount API
 *
 * 后端已完成 model_v2 → model 扁平化转换，前端直接使用
 */

import { get } from "./request";

// ============================================================
// 类型定义（匹配后端 model 响应）
// ============================================================

export interface ServiceAccountItem {
  name: string;
  namespace: string;
  secretsCount: number;
  imagePullSecretsCount: number;
  automountServiceAccountToken?: boolean;
  createdAt: string;
  age: string;
}

export interface ServiceAccountDetail {
  name: string;
  namespace: string;
  secretsCount: number;
  imagePullSecretsCount: number;
  automountServiceAccountToken?: boolean;
  secretNames?: string[];
  imagePullSecretNames?: string[];
  createdAt: string;
  age: string;
  labels?: Record<string, string>;
  annotations?: Record<string, string>;
}

// ============================================================
// 响应类型（内部使用）
// ============================================================

interface ListResponse<T> {
  message: string;
  data: T[];
  total: number;
}

interface DetailResponse<T> {
  message: string;
  data: T;
}

// ============================================================
// API 查询参数（内部使用）
// ============================================================

interface ClusterResourceParams {
  cluster_id: string;
  namespace?: string;
}

// ============================================================
// API Functions
// ============================================================

export function getServiceAccountList(params: ClusterResourceParams) {
  return get<ListResponse<ServiceAccountItem>>("/api/v2/service-accounts", params);
}

export function getServiceAccountDetail(params: { ClusterID: string; Namespace: string; Name: string }) {
  return get<DetailResponse<ServiceAccountDetail>>(
    `/api/v2/service-accounts/${encodeURIComponent(params.Name)}`,
    { cluster_id: params.ClusterID, namespace: params.Namespace }
  );
}
